tests/Contacts/ContactSortingTest.m

```diff
@@ -0,0 +1,119 @@
+{
+    NSComparator comparator = [Contact comparatorSortingNamesByFirstThenLast:NO];
+    NSArray<Contact *>*sortedContacts = [self.class contactArrayForNames:@[@[@"Mary Jo", @"Catlett"],
+                                                                           @[@"Daisy", @"Chain"],
+                                                                           @[@"", @"Daisy"],
+                                                                           @[@"Daisy", @"Duke"],
+                                                                           @[@"Mary", @"Oliver"],
+                                                                           @[@"Adam", @"Smith"],
+                                                                           @[@"James", @"Smith"],
+                                                                           @[@"Kevin", @"Smith"],
+                                                                           @[@"James", @"Van"],
+                                                                           @[@"James", @"Van Der Beek"],
+                                                                           @[@"Adam", @"West"],
+                                                                           @[@"Mae", @"West"],
+                                                                           ]];
+    NSUInteger numContacts = sortedContacts.count;
+    
+    for (NSUInteger i = 0; i < 20; i++) {
+        NSArray *shuffledContacts = [self.class shuffleArray:sortedContacts];
+        NSArray *resortedContacts = [shuffledContacts sortedArrayUsingComparator:comparator];
+        for (NSUInteger j = 0; j < numContacts; j++) {
+            Contact *a = sortedContacts[j];
+            Contact *b = resortedContacts[j];
+            BOOL correct = ([a.firstName isEqualToString:b.firstName] && [a.lastName isEqualToString:b.lastName]);
+            if (!correct) {
+                XCTAssert(@"Contacts failed to sort names by last, first");
+                break;
+            }
+        }
+    }
+}
+
++ (NSArray<Contact *> *)contactArrayForNames:(NSArray<NSArray<NSString *>*>*)namePairs
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    ABRecordID fakeRecordId = 0;
+#pragma clang diagnostic pop
+    NSMutableArray<Contact *>*contacts = [[NSMutableArray alloc] initWithCapacity:namePairs.count];
+    for (NSArray<NSString *>*namePair in namePairs) {
+        Contact *c = [[Contact alloc] initWithContactWithFirstName:namePair[0]
+                                                       andLastName:namePair[1]
+                                           andUserTextPhoneNumbers:@[]
+                                                          andImage:nil
+                                                      andContactID:fakeRecordId++];
+        [contacts addObject:c];
+    }
+    
+    return [NSArray arrayWithArray:contacts]; // Return an immutable for good hygene
```

Comment:
`[contacts copy]` is a more idiomatic way to return an immutable object from it's mutable variant.